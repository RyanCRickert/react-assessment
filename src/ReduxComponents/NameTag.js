import React, { useState } from "react";

const NameTag = ({ name, language }) => {
  const [color, updateColor] = useState(1);

  const changeColor = () => {
    if(color < 4 && color > 0) {
      updateColor(color + 1)
    } else (
      updateColor(1)
    )
  }

  return (
    <div onClick={changeColor} className="Name-tag_container">
      <div className={`Name-tag_greeting${color}`}>{language}</div>
      <div className="Name-tag_person">{name}</div>
      <div className={`Name-tag_bottom${color}`} />
    </div>
  )
}

export default NameTag;

