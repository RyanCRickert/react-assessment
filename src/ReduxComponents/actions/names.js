export const changeNames = (names = "") => ({
  type: "SET_NAMES",
  names
})