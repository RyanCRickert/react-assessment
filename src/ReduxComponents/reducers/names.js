const namesReducerDefaultState = {
  names: ""
}

export default (state = namesReducerDefaultState, action) => {
  switch(action.type) {
    case "SET_NAMES":
      return {
        ...state,
        names: action.names
      }
    default:
      return state;
  }
}