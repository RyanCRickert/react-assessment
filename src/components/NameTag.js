import React from "react";

const NameTag = ({ name, language, color, changeColor }) => {
  return (
    <div onClick={changeColor} className="Name-tag_container">
      <div className={`Name-tag_greeting${color}`}>{language}</div>
      <div className="Name-tag_person">{name}</div>
      <div className={`Name-tag_bottom${color}`} />
    </div>
  )
}

export default NameTag;