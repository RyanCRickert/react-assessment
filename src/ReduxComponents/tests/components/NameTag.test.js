import React from "react";
import { shallow } from "enzyme";
import { NameTag } from "../../NameTag";
import names from "../fixtures/names";

test("should render NameTag with a name", () => {
  const wrapper = shallow(<NameTag {...names[0]} />)
  expect(wrapper).toMatchSnapshot();
});