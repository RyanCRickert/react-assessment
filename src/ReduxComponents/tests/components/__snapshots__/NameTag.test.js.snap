// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NameTag with a name 1`] = `
<div
  className="Name-tag_container"
  onClick={[Function]}
>
  <div
    className="Name-tag_greeting1"
  />
  <div
    className="Name-tag_person"
  />
  <div
    className="Name-tag_bottom1"
  />
</div>
`;
